client/app/components/account_config.coffee

```diff
@@ -0,0 +1,382 @@
+            smtpPassword: allowEmpty: true
+            draftMailbox: allowEmpty: true
+            sentMailbox: allowEmpty: true
+            trashMailbox: allowEmpty: true
+            accountType: allowEmpty: true
+
+
+    getInitialState: ->
+        return @accountToState null
+
+
+    # Do not update component if nothing has changed.
+    shouldComponentUpdate: (nextProps, nextState) ->
+        isNextState = _.isEqual nextState, @state
+        isNextProps = _.isEqual nextProps, @props
+        return not (isNextState and isNextProps)
+
+
+    componentWillReceiveProps: (props) ->
+        # prevents the form from changing during submission
+        if props.selectedAccount? and not props.isWaiting
+            @setState @accountToState props
+
+        else
+
+            if props.error?
+
+                if props.error.name is 'AccountConfigError'
+                    errors = {}
+                    field = props.error.field
+
+                    if field is 'auth'
+                        errors.login = t 'config error auth'
+                        errors.password = t 'config error auth'
+
+                    else
+                        errors[field] = t 'config error ' + field
+
+                    @setState errors: errors
+            else
+
+                if not props.isWaiting and not _.isEqual(props, @props)
+                    @setState @accountToState null
+
+
+    render: ->
+        mainOptions = @buildMainOptions()
+        mailboxesOptions = @buildMailboxesOptions()
+        titleLabel = @buildTitleLabel()
+        tabParams = @buildTabParams()
+
+        Container
+            id: 'mailbox-config'
+            key: "account-config-#{@props.selectedAccount?.get 'id'}"
+        ,
+            Title text: titleLabel
+            if @props.tab?
+                Tabs tabs: tabParams
+            if not @props.tab or @props.tab is 'account'
+                AccountConfigMain mainOptions
+            else
+                AccountConfigMailboxes mailboxesOptions
+
+
+    # Ask to main layout manager to display error as notification toasters.
+    renderError: (errors) ->
+        LayoutActionCreator.alertError t 'account errors'
+
+
+    # Build options shared by both tabs.
+    buildMainOptions: (options) ->
+
+        options =
+            isWaiting: @props.isWaiting
+            selectedAccount: @props.selectedAccount
+            validateForm: @validateForm
+            onSubmit: @onSubmit
+            onBlur: @onFieldBlurred
+            errors: @state.errors
+            checking: @state.checking
+
+        options[field] = @linkState field for field in @_accountFields
+
+        return options
+
+
+    # Build options required by mailbox tab.
+    buildMailboxesOptions: (options) ->
+
+        options =
+            error: @props.error
+            errors: @state.errors
+            onSubmit: @onSubmit
+
+        options[field] = @linkState field for field in @_mailboxesFields
```

Comment:
I don't know, it's just a copy paste of the previous code. I don't really understand what does this line.